### PR TITLE
Adding @mapbox/vector-tile + @mapbox/point-geometry

### PR DESCRIPTION
**Pull request by webberig**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by HarelM — APPROVED** [review]

File: types/mapbox__point-geometry/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    mag: () => number;
+    equals: (other: Point) => boolean;
+    dist: (p: Point) => number;
+    distSqr: (p: Point) => number;
+    angle: () => number;
+    angleTo: (b: Point) => number;
+    angleWith: (b: Point) => number;
+    angleWithSep: (x: number, y: number) => number;
+
+    static convert: (a: number[] | Point) => Point;
```
Comment (peterblazejewicz) [inline types/mapbox__point-geometry/index.d.ts:31]: what about this line:
```suggestion
    static convert<T extends unknown>(a: T): T extends number[] ? Point : T extends Point ? Point : T;
```
`a` is required, but depending on input, output is different:
https://github.com/mapbox/point-geometry/blob/master/index.js#L296. When T is a point, T could be returned to outline identity.

Comment (HarelM) [reply types/mapbox__point-geometry/index.d.ts:31]: That's probably more accurate... Thanks!

**Review by HarelM — APPROVED** [review]

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@webberig thx!

**Comment by webberig** [comment]
Ready to merge

**Pull request merged**